discovery-infra/install_cluster.py

```diff
@@ -208,8 +208,10 @@ def run_installation_flow_kube_api(
         kubeconfig_path,
 ):
     log.info("Approving agents")
-    for agent in cluster_deployment.wait_for_agents(nodes_number):
+    agents = cluster_deployment.wait_for_agents(nodes_number)
+    for agent in agents:
         agent.approve()
+        agent.wait_for_agents_to_install(agents, nodes_number)
 
```

Comment:
All the agents should be approved and only then monitored. Otherwise the installation process won't start